mlir/lib/Dialect/Rock/Transforms/ConvToGemm.cpp

```diff
@@ -145,22 +145,33 @@ Type getResultType(Operation *convOp, Value outArg) {
   return nullptr;
 }
 
+static int64_t getUtilityVectorizationLen(ShapedType shape) {
```

Comment:
Could you explain a bit the logic here? Why `kMaxVectorOpLen=4`? Is this the same for every architecture?  If it is, is there any design doc you can cite? Does this mean every thread only computes a single element? What about more than one element per thread?

Reply:
MaxVectorOpLen is 4 words because there're no load ops on anything we target that's more than four words long. You'll see similar magic 4's in the global_load and global_store patterns.

Each threads computes `elemsPerThreads` elements at most, which is currently fixed to `512`